### PR TITLE
Tips: communication between sibling components

### PR DESCRIPTION
**Pull request by ashwin01**
Added tips on how can communication be established between sibling components, with an example. Please do review it, thanks!

### CODE REVIEW

**Comment by jimfb** [comment]
Reading through this, it takes a long time to figure out what the doc is trying to say.  I wonder if we really need to provide a full-fledged example for sibling communication.  Would it be sufficient to say:
 > For communication between two sibling components, you can usually create parent component that coordinates between the two children.  The parent component can receive messages (events/callbacks) from one child, update it's internal state, and pass the updated state/props to the other child.

I feel like that is simple and concise, and provides the necessary information.  Realistically, when people start hitting situations where they need to pass information outside of a parent-child relationship, they should probably start learning about Flux.

That's my two cents.  @zpao, thoughts?  @spicyj?

File: docs/tips/14-communicate-between-components.md [hunk 1]
```diff
@@ -39,3 +39,72 @@ ReactDOM.render(
 Notice the use of `bind(this, arg1, arg2, ...)`: we're simply passing more arguments to `handleClick`. This is not a new React concept; it's just JavaScript.
 
 For communication between two components that don't have a parent-child relationship, you can set up your own global event system. Subscribe to events in `componentDidMount()`, unsubscribe in `componentWillUnmount()`, and call `setState()` when you receive an event. [Flux](https://facebook.github.io/flux/) pattern is one of the possible ways to arrange this.
+
+
+For communication between two components that don't have a parent-child relationship, you can set up your own global event system. Subscribe to events in `componentDidMount()`, unsubscribe in `componentWillUnmount()`, and when you receive an event, call `setState()`.
```
Comment (zpao) [inline docs/tips/14-communicate-between-components.md]: I don't think you meant to include this - it's part of the paragraph above.

**Comment by zpao** [comment]
Yea, I think I agree with @jimfb. I think a full-fledged example isn't quite helpful and (maybe it's just me) made me want to write it differently. Really what you've done is parent-child communication. The siblings still don't know anything about each other and it's not a great idea to start depending on that, even if they did.

**Comment by ashwin01** [comment]
Yes, I agree that the example is rather verbose. Would reducing the lines of explanation make it easier?

Basically I thought this approach could be useful when the parent has lots of nested components, and updating the state in the parent would make React run `render` on all the child components, which could be avoided. I do agree that it isn't exactly sibling communication in the strictest definition, and they use the parent to pass information.

There might be a case where your child component has some state, hence passing data to it via props wouldn't work. You might still set the state of the child from props, but seems like that is an anti-pattern in React, correct?

**Comment by ashwin01** [comment]
@zpao My bad, I didn't mean to include that line. Thanks for pointing it out! :-)

**Comment by ashwin01** [comment]
Hi, may I know the status of this P.R ?

**Comment by jimfb** [comment]
@ashwin01 We're probably not going to accept this PR as-is; it is too complicated.  Changing the wording around the example would not be sufficient - the problem is the example.  A couple of sentences could cover this topic, without introducing so much complicated logic for the user to read/understand.  As per my https://github.com/facebook/react/pull/5647#issuecomment-164043976 above, the changes in this PR could be simplified by replacing all these changes with the sentences from my comment.

**Pull request closed without merging**